Add workshop files (#3)
pyladiesams/data-pipelines-with-dbtcore-snowflake-sep2025: workshop/workshop_setup.md

# Workshop requirements setup

These setup instructions will guide you through the core steps required to use dbt Core with Snowflake:
- Step 1. [Create a Snowflake trial account](#step-1-create-a-trial-account-with-snowflake)
- Step 2. [Create a PROD_RAW database in Snowflake and load sample data](#step-2-create-a-prod_raw-database-in-snowflake-and-load-sample-data)
- Step 3. [Install dbt Core locally](#step-3-install-dbt-core)
- Step 4. [Configure your dbt environment](#step-4-configure-your-dbt-environment)

## Step 1. Create a trial account with Snowflake

>  - Go to the [Snowflake sign up page](https://signup.snowflake.com/) to create a trial account. The trial account is **valid for 30 days** and includes $400 worth of free usage. You new accoutn should have the follwoign settings:

Reviewer: ```suggestion
 - Go to the [Snowflake sign up page](https://signup.snowflake.com/) to create a trial account. The trial account is **valid for 30 days** and includes $400 worth of free usage. You new account should have the following settings:
```

    - Snowflake edition: Standard
    - Cloud provider: any cloud provider will work for this workshop

>     - Region: any siutable region based on your location

Reviewer: ```suggestion
    - Region: any suitable region based on your location
```

   ```
 - The final step is to install dbt and Snowflake adapter:

   ```bash
   pip install dbt-core dbt-snowflake
   ```
   Run the following command to confirm that dbt was installed correctly:

   ```bash
   dbt --version
   ```

  Here is the [dbt core + Snowflake documentation](https://docs.getdbt.com/docs/core/connect-data-platform/snowflake-setup#user--password-authentication) for more details.

## Step 4. Configure your dbt Environment

For this workshop, we'll set up a local profiles.yml file to connect to Snowflake. This project includes a pre-configured `profiles.yml` file located in the `workshop/gallery_project/dbt_profiles/profiles.yml` directory. This means that you only need to update your `.env` file with correct credentials.

We'll use environment variables to securely manage our Snowflake connection details. This approach prevents sensitive information from being stored directly in configuration files.

1. Create a `.env` file in the project root (workshop/gallery_project) to store your environment variables:

   ```bash
   touch .env
   ```

2. Add your Snowflake connection details to the `.env` file (replace with your actual values):

  ```bash

>   export DBT_PROFILES_DIR=dbt_profiles

Reviewer: ```suggestion
  export DBT_PROFILES_DIR=workshop/gallery_project/dbt_profiles
```

I had to change the path here.

Author: That's a good catch: you can either change the path, or cd into the project folder to run your models. I'll add a note on that to the setup document.

>   export SNOWFLAKE_ACCOUNT_ID='your-account-id'  # e.g., xy12345.us-east-1

Reviewer: ```suggestion
  export DBT_PROJECT_DIR=workshop/gallery_project
  export SNOWFLAKE_ACCOUNT_ID='your-account-id'  # e.g., xy12345.us-east-1
```

I had to add an additional line for the PROJECT_DIR.

Author: It depends on which folder you are running your dbt commands from. I'll update the instructions to make it clear.

Reviewer: Makes sense!

  export SNOWFLAKE_USERNAME='your-username'
  export SNOWFLAKE_PASSWORD='your-password'
  export USER_ROLE='SYSADMIN'

>   export DATABASE='DEV_BRONZE'

Reviewer: ```suggestion
  export DATABASE='DEV_BRONZE'
```
Should this be aligned with the database name set above?

Author: DEV_BRONZE is the default database for the profile. If dbt cannot find the correct reference for where to store new tables or views, it will write them to this database.

  export WAREHOUSE='COMPUTE_WH'

>   export SCHEMA='DBT_DEV'

Reviewer: ```suggestion
  export SCHEMA='DBT_WORKSHOP'
```

Should this be aligned with the schema set above?

Author: This is a different schema in the DEV_BRONZE database, so it's correct. The DBT_WORKSHOP schema is in the PROD_RAW database (i.e., the original source data). Let me rename DBT_WORKSHOP schema to avoid the confusion.
